test/jdk/valhalla/valuetypes/IsIdentityClassTest.java

```diff
@@ -44,48 +41,51 @@
 import java.util.List;
 import java.util.Set;
 
+import jdk.internal.misc.PreviewFeatures;
 import jdk.internal.value.ValueClass;
 
 import static jdk.test.lib.Asserts.*;
 
 public class IsIdentityClassTest {
 
-    static boolean UseArrayFlattening = false;
-    static {
+    private static void assertFalseIfPreview(boolean condition, String msg) {
+        if (PreviewFeatures.isEnabled()) {
+            assertFalse(condition, msg);
+        } else {
+            assertTrue(condition, msg);
+        }
+    }
+
+    @Test
+    void testIsIdentityClass() {
         RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
         List<String> arguments = runtimeMxBean.getInputArguments();
-        UseArrayFlattening = !arguments.contains("-XX:-UseArrayFlattening");
+        boolean UseArrayFlattening = !arguments.contains("-XX:-UseArrayFlattening");
```

Comment:
Optional comments:

I wonder if it's more usually to obtain flags with serviceability, or explicitly pass -D properties...

Also for these global states, it's better to capture them with a `@Setup static void setup()` method and store in static fields, especially if multiple tests use these states.

Reply:
I copied this out of a hotspot valhalla test.  If I remove the flattening tests, I suppose I won't need this anymore.

Reply:
There might be a better way to find this out though.  We use this in the Hotspot runtime valhalla tests, but those tests don't use junit, so @things aren't used.  For Hotspot, we're passing -XX flags for now and not using properties.